config/op-consumer/public/util.js

```diff
@@ -89,14 +89,38 @@ function partition(arr, fn) {
 
 /**
  * Send triples to reasoning service for conversion
- *
  */
 function transformTriples(fetch, triples) {
   return operationWithRetry(mainConversion(fetch, triples), 0,
     MAX_REASONING_RETRY_ATTEMPTS, SLEEP_TIME_AFTER_FAILED_REASONING_OPERATION);
 }
 
 
+/**
+ * Remove triples which were added to the context but not present in the original triples
+ * Triples which are result of the transformation will be kept.
+ * @param {*} transformedTriples
+ * @param {*} originalTriples
+ * @param {*} triplesWithContext
+ */
+function removeContextTriples(transformedTriples, originalTriples, triplesWithContext) {
```

Comment:
Not sure I understand, based on the input. To find the transformed statement, I would expect the following operation:
`let A = {transformed statements, transformed context} # The mega blob of transformed statements`
`let B = {transformed context}  # Blob only transformed context statements`
`A\B = {transformed statements}`
However, *how I understand* here you seem to assume that only what is not in the original context are statements needed to further apply on the database. 
But what if e.g the consuming app works on `besluit:Eenheid` and not on `besluit:Bestuurseenheid`, but you need this information to update the `skos:prefLabel`? 

I really might be mistaken, on a Friday afternoon.

Reply:
It's indeed tricky. 
- The current solution would cause problems if (not currently the case) a class is derived based on multiple possible properties for example.
- One issue is that often transformations without context yield no results at all - because the graph patterns do not match due to the missing context.
- After quite some thinking and thinkering, I think this issue with the scope of deletes could require so much logic at the step where the context and things derived from the context should be removed from the reasoning result, that it starts to make less sense to separate the reasoning logic into the rules. As much of it is duplicated in the latter part. One part where it might still make sense (arguably) is the initial sync, since the transformation through reasoning is only a matter of a few seconds.